lib/puppet/version.rb

```diff
@@ -5,6 +5,11 @@
 # The version can be set programmatically because we want to allow the
 # Raketasks and such to set the version based on the output of `git describe`
 
+# Seems to be the minimum to get a vendored library without adding extra dependencies.
+vendor_dir = File.join([File.dirname(File.expand_path(__FILE__)), "vendor"])
+load "#{vendor_dir}/load_semantic_puppet.rb"
```

Comment:
Ouch. Didn't anticipate this and I don't like it. The proper way to load the vendored stuff is to use `Puppet::Vendor.load_vendored` so this seems a bit patchy. Using the proper way will however not work since it requires `Puppet.debug` which is instantiated much later. Based on that, I'd rather go back to the previous solution with split/join. @MosesMendoza , do you agree with this?